code/ui/blocks/src/blocks/Story.stories.tsx

```diff
@@ -13,8 +15,84 @@ export default meta;
 
 type Story = StoryObj<typeof meta>;
 
-export const BasicOf: Story = {
+export const Of: Story = {
   args: {
     of: BooleanStories.Undefined,
   },
 };
+
+export const OfWithMeta: Story = {
+  args: {
+    of: BooleanStories.True,
+    meta: BooleanStories.default,
+  },
+};
+
+const isBlocksOnly = import.meta.env.STORYBOOK_BLOCKS_ONLY === 'true';
```

Comment:
@IanVS I've experienced some annoying/surprising things here that I would like your thoughts on, related to Vite:

1. Reading from `process.env.STORYBOOK_BLOCKS_ONLY` works in dev, but not when built (`process is undefined`)
2. Reading from `import.meta.env.STORYBOOK_BLOCKS_ONLY` works in both dev and prod, but only in stories. In `main.ts` it doesn't even compile - `import.meta.env is undefined` -  so I have to stick to `process.env` there. 
3. using `import.meta.env` directly at line 35 below throws an error (see below), but defining it top level seem to work. (this might be expected?)

```
[storybook:inject-export-order-plugin] Parse error @:100:174
file: /Users/jeppe/dev/chromatic/storybook/code/ui/blocks/src/blocks/Story.stories.tsx
info => Loading presets
ERR! Error: Parse error @:100:174
ERR!     at parse (/Users/jeppe/dev/chromatic/storybook/code/node_modules/es-module-lexer/dist/lexer.cjs:1:402)
```

I wonder how much of this is fixable and how much of it we have to document as gotchas in https://storybook.js.org/docs/7.0/react/configure/environment-variables

Reply:
Hi, without digging into it too much, here's what I think:

1) Using `process.env.anything` should not work in the browser, unless it has been added to the `define` config for vite (https://vitejs.dev/config/shared-options.html#define).  I'm not sure why you're seeing it work in dev, I'd need to dig in a bit.

2) This is expected.  `main.js` runs in node, not the browser, so it uses node APIs like `process.env`.

3) I opened an issue in es-module-lexer: https://github.com/guybedford/es-module-lexer/issues/142.  We could try finding an alternative if it doesn't seem like it will be addressed.

Reply:
@JReinhold can you help me reproduce issue 3?  I pulled down the branch, changed `isBlocksOnly` below to `import.meta.env.STORYBOOK_BLOCKS_ONLY === 'true'`, ran `yarn task --task compile --start-from install` then `yarn storybook:blocks`, and it was fine, and the `id` story rendered without errors.  I'm guessing I'm doing something wrong, though.

Reply:
> 2. This is expected.  `main.js` runs in node, not the browser, so it uses node APIs like `process.env`.

Yeah that's fair, I'm just surprised that node doesn't support `import.meta.env`, but that's not our fault. All we can do is document it.

> can you help me reproduce issue 3?

Sorry, I forgot the crucial detail that it only fails when building, not in dev.

Reply:
Jeppe and I dug into this a bit, and there are a few things going on:

1) The main cause of failure here is that `import.meta.env.STORYBOOK_BLOCKS_ONLY` is being replaced by vite to `"true"` (including the double-quotes).  This closes out the starting double-quote, and causes a parse error.  
1) The reason that the error is different between `storybook:blocks:build` and `storybook:ui:build`, is that `STORYBOOK_BLOCKS_ONLY` is undefined in storybook:ui:build.

I've opened an issue to track this: https://github.com/storybookjs/storybook/issues/19830.